resolvers/patient.js

```diff
@@ -1,10 +1,65 @@
+/* eslint-disable prettier/prettier */
+
 'use strict';
 
+const { PubSub, withFilter } = require('apollo-server');
+const pubsub = new PubSub();
 const db = require('../models');
 const { Roles } = require('../utils/enum');
 const validators = require('../utils/validators');
 
+const PATIENT_ADDED = 'PATIENT_ADDED';
+const PATIENT_UPDATED = 'PATIENT_UPDATED';
+const PATIENT_DELETED = 'PATIENT_DELETED';
+const PATIENT_RESTORED = 'PATIENT_RESTORED';
+
 const patientResolvers = {
+  Subscription: {
+    patientAdded: {
+      subscribe: withFilter(
+        () => pubsub.asyncIterator([PATIENT_ADDED]),
+        (payload, variables) => {
+          return (
+            payload.patientAdded.collectionPointId ===
+            parseInt(variables.collectionPointId)
+          );
+        }
+      ),
+    },
+    patientUpdated: {
+      subscribe: withFilter(
+        () => pubsub.asyncIterator([PATIENT_UPDATED]),
+        (payload, variables) => {
+          return (
+            payload.patientUpdated.collectionPointId ===
+            parseInt(variables.collectionPointId)
+          );
+        }
+      ),
+    },
+    patientRestored: {
```

Comment:
@petorsong just want to confirm that it's ok for this to only consider patients that are "restored" through the `restorePatient` patient mutation and ignores using `editPatient` to change the status of a patient from `DELETED` to something else.

Reply:
yep - we're basically assuming that the frontend will never send us requests of that sort (which isn't ideal) but this is fine for now